File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F05.P2.zig
const std = @import("std");

fn probe() i32 {
// BEGIN PROBE F05.P2
const put = struct {
    fn call(out: *i32) void {
        out.* = 12;
    }
}.call;
var cell: i32 = 3;
put(&cell);
return cell;
// END PROBE F05.P2
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{probe()});
    try out.interface.flush();
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F05.P2.rs
fn probe() -> i32 {
// BEGIN PROBE F05.P2
fn put(out: &mut i32) { *out = 12; }
let mut cell = 3;
put(&mut cell);
cell
// END PROBE F05.P2
}

fn main() {
    println!("{}", probe());
}
